onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -1035,6 +1115,52 @@ TensorrtExecutionProvider::GetCapability(const GraphViewer& graph,
 
   // Construct subgraph capability from node list
   std::vector<std::unique_ptr<ComputeCapability>> result;
+
+  // Handle the case where the graph is subgraph of control flow op.
+  // The purpose is to make control flow op as well as its subgraphs run on TRT.
+  // Here we need to check whether subgraph is fully supported by TRT and don't fuse the nodes of the subgraph until control flow op level.
+  if (IsSubGraphOfControlFlowOp(graph) && IsSubGraphFullySupported(supported_nodes_vector, number_of_ort_nodes)) {
+    const std::vector<NodeIndex>& node_index = graph.GetNodesInTopologicalOrder();
+    bool all_subgraphs_are_supported = true;
+
+    // "If" control flow op has two subgraph bodies, "then" body and "else" body respectively.
+    // Check its parent node's another subgraph to see whether that subgraph is also fully supported by TRT.
```

Comment:
What's the reason that both else and then subgraphs must be fully supportedT? If one is supported and the other one is not, can the first one run in TRT?

Reply:
Discussed offline.
Any portion of control flow op's subgraphs that can be run on TRT, will still be assigned to TRT EP and run on TRT.